### ISSUE TITLE
W: The repository 'http://security.ubuntu.com/ubuntu xenial-security Release' does not have a Release file.

### ISSUE THREAD

**Issue by sholy123** [NVIDIA/nvidia-docker#960]
W: The repository 'http://security.ubuntu.com/ubuntu xenial-security Release' does not have a Release file.
W: The repository 'http://archive.ubuntu.com/ubuntu xenial Release' does not have a Release file.
W: The repository 'http://archive.ubuntu.com/ubuntu xenial-updates Release' does not have a Release file.
W: The repository 'http://archive.ubuntu.com/ubuntu xenial-backports Release' does not have a Release file.
E: Failed to fetch http://security.ubuntu.com/ubuntu/dists/xenial-security/main/binary-amd64/Packages  Connection failed [IP: 91.189.88.149 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/xenial/main/binary-amd64/Packages  Connection failed [IP: 91.189.88.161 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/xenial-updates/main/binary-amd64/Packages  Connection failed [IP: 91.189.91.23 80]
E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/xenial-backports/main/binary-amd64/Packages  Connection failed [IP: 91.189.88.152 80]
E: Some index files failed to download. They have been ignored, or old ones used instead.

when i building the dockerfile, i meet the problem.when i run apt-get update,the problem comes.how can i fix this probelm?

**Comment by mercury00** [comment]
The problem is `connectino failed [IP 91.189.88.161.80]` - this isn't a problem with the repo missing a release file, it's a problem with your routing/internet. 

You'll need to inspect your docker configuration and host configuration to be sure that 1) your computer has access to security.ubuntu.com and archive.ubuntu.com (try a `curl -s -L http://archive.ubuntu.com/ubuntu/dists`), and that your docker container is able to route to the internet as well.

To summarize, you have a networking problem, your host or your container are not able to route to the ip address that ubuntu.com resolves to (91.189.88.149 for instance), whether by firewall or misconfigured routes or ip address configurations or missing interfaces or something else. Hope this helps you diagnose the issue.

**Comment by RenaudWasTaken** [comment]
Thanks for helping @mercury00 !

**Comment by pieroir** [comment]
**get sources.list from ** https://repogen.simplylinux.ch/generate.php

and replace to your /etc/apt/sources.list with the downloaded file . next

sudo apt update
sudo apt dist-upgrade
